proxy/core/connection/pool.py

```diff
@@ -64,6 +60,7 @@ class ConnectionPool:
     def __init__(self) -> None:
         # Pools of connection per upstream server
         self.pools: Dict[Tuple[str, int], Set[TcpServerConnection]] = {}
+        self.handshake_pools: Dict[Tuple[str, int], Dict[int, bytes]] = {}
```

Comment:
A doc-string about this data-structure will be nice.  May be I can send out contributions related to docs directly into your PR.

Reply:
We can remove this statement too, it not needed anymore

Reply:
Is this not being used?  If not let's comment or remove any unused bit.